### DISCUSSION TITLE
Define languagues in settings

### DISCUSSION THREAD

**Discussion by curquiza** [meilisearch/product#702] [Feedback & Feature Proposal]
Following a discussion with @ManyTheFish 

We encounter issues where letting the users set the language(s) of their dataset would solve completely or partially their issues
- Japanese users: https://github.com/meilisearch/meilisearch/issues/4162 -> currently our Japanese users don't use Meilisearch but the alternative image Many provides in https://github.com/meilisearch/meilisearch/pull/3882
- https://github.com/meilisearch/meilisearch/issues/4133 -> Meilisearch could behave for sorting based on language detection

**Comment by uintaam** [comment]
This would also help you if you decided to add Japanese Transliteration, as it's the only language I know where the written reading is essential to search.  i.e., If the value in the dictionary is 日本語は楽しい, it would be ideal for someone searching with the hiragana or katakana versions of 日本語 (にほんご, ニホンゴ) or 楽しい (たのしい,タノシイ) to see the results

**Comment by curquiza** [comment]
More information about a user having different language in the same document: https://github.com/meilisearch/meilisearch/issues/4162#issuecomment-1818314376

**Comment by ManyTheFish** [comment]
Related to [A user Mixing several Languages in the same index](https://github.com/meilisearch/meilisearch/discussions/4324).

**Comment by uintaam** [comment]
Another related issue https://github.com/meilisearch/meilisearch/issues/4561

**Comment by ManyTheFish** [comment]
This issue has been forgotten,

the [localized-attributes setting](https://www.meilisearch.com/docs/reference/api/settings#localized-attributes) added in 2024 should solve this issue,

Closing it!
